pkg/csi/cinder/controllerserver.go

```diff
@@ -1038,18 +1036,20 @@ func (cs *controllerServer) ControllerExpandVolume(ctx context.Context, req *csi
 	}, nil
 }
 
-func getTopology(vol *volumes.Volume, topologyReq *csi.TopologyRequirement, ignoreVolumeAZ bool) []*csi.Topology {
+func getTopology(vol *volumes.Volume, topologyReq *csi.TopologyRequirement, withTopology bool, ignoreVolumeAZ bool) []*csi.Topology {
 	var accessibleTopology []*csi.Topology
-	if ignoreVolumeAZ {
-		if topologyReq != nil {
-			accessibleTopology = topologyReq.GetPreferred()
-		}
-	} else {
-		// NOTE(stephenfin): We retrieve the AZ from the created volume rather than
-		// using the value we provided in our request since these can differ due to
-		// Cinder's '[DEFAULT] allow_availability_zone_fallback' option.
-		accessibleTopology = []*csi.Topology{
-			{Segments: map[string]string{topologyKey: vol.AvailabilityZone}},
+	if withTopology {
+		if ignoreVolumeAZ {
+			if topologyReq != nil {
+				accessibleTopology = topologyReq.GetPreferred()
+			}
+		} else {
+			// NOTE(stephenfin): We retrieve the AZ from the created volume rather than
+			// using the value we provided in our request since these can differ due to
+			// Cinder's '[DEFAULT] allow_availability_zone_fallback' option.
+			accessibleTopology = []*csi.Topology{
+				{Segments: map[string]string{topologyKey: vol.AvailabilityZone}},
+			}
```

Comment:
IIUC, these changes originate from the topology defined on the CSINode. With this change we're (optionally) ignoring topology requested by the Node and only observing topology directly requested on the PV via the StorageClass (or PVC?).

This makes sense to me.

Didn't you also discuss allowing `--with-topology=false` to disable setting topology on the CSINode altogether? I was expecting to see that in this PR.